crypto/fipsmodule/ssh/kdf.c

```diff
@@ -0,0 +1,105 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+// Based on OpenSSL 3.x SSHKDF().
+
+#include <openssl/digest.h>
+#include <openssl/mem.h>
+
+#include "../service_indicator/internal.h"
+#include "../../internal.h"
+
+int SSHKDF(const EVP_MD *evp_md,
+           const uint8_t *key, size_t key_len,
+           const uint8_t *xcghash, size_t xcghash_len,
+           const uint8_t *session_id, size_t session_id_len,
+           char type,
```

Comment:
Shouldn't this method do some checks to ensure `type` is one of the `EVP_KDF_SSHKDF_TYPE`?

Reply:
That's a good idea; I wasn't sure those were the only valid choices, but they do seem to be the only choices available in OpenSSL. I'll add checks and an additional unit test.